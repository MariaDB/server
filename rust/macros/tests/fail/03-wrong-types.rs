/*
 * Verify our added check for identifier-type mismatch
 */

include!("../include.rs");

register_plugin! {
    TestPlugin,
    ptype: PluginType::MariaEncryption,
    name: "debug_key_management",
    author: "Trevor Gross",
    description: "Debug key management plugin",
    license: License::Gpl,
    maturity: Maturity::Experimental,
    version: "0.1",
    init: TestPlugin,
    encryption: false,
    variables: [
        SysVar {
            ident: _SYSVAR_ATOMIC,
            vtype: SysVarConstString,
            name: "test_sysvar",
            description: "this is a description",
            options: [SysVarOpt::ReadOnly, SysVarOpt::NoCmdOpt],
            default: "default value"
        }
    ]
}

fn main() {}
