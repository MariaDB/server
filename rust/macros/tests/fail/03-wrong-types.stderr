error[E0308]: mismatched types
  --> tests/fail/03-wrong-types.rs:7:1
   |
7  | / register_plugin! {
8  | |     TestPlugin,
9  | |     ptype: PluginType::MariaEncryption,
10 | |     name: "debug_key_management",
...  |
27 | |     ]
28 | | }
   | |_^ expected `&SysVarConstString`, found `&AtomicI32`
   |
   = note: expected reference `&'static mariadb::plugin::SysVarConstString`
              found reference `&AtomicI32`
   = note: this error originates in the macro `register_plugin` (in Nightly builds, run with -Z macro-backtrace for more info)
